connect-location/src/main/java/com/ifttt/location/GeofenceProvider.java

```diff
@@ -0,0 +1,115 @@
+package com.ifttt.location;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import androidx.annotation.RequiresPermission;
+import com.google.android.gms.awareness.Awareness;
+import com.google.android.gms.awareness.FenceClient;
+import com.google.android.gms.awareness.fence.FenceQueryRequest;
+import com.google.android.gms.awareness.fence.FenceUpdateRequest;
+import com.google.android.gms.awareness.fence.LocationFence;
+import com.ifttt.connect.LocationFieldValue;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+final class GeofenceProvider {
```

Comment:
What is the reason behind separating this class from ConnectLocation?

Reply:
just to separate the geofence implementation from the main `ConnectLocation` class which recives connect button state change callback, parsing the user features, polling..